src/fstab-generator/fstab-generator.c

```diff
@@ -215,6 +215,42 @@ static int write_requires_after(FILE *f, const char *opts) {
         return 0;
 }
 
+static int write_requires_before(FILE *f, const char *opts) {
+        _cleanup_strv_free_ char **names = NULL, **units = NULL;
+        _cleanup_free_ char *res = NULL;
+        char **s;
+        int r;
+
+        assert(f);
+        assert(opts);
+
+        r = fstab_extract_values(opts, "x-systemd.before", &names);
+        if (r < 0)
+                return log_warning_errno(r, "Failed to parse options: %m");
+        if (r == 0)
+                return 0;
+
+        STRV_FOREACH(s, names) {
+                char *x;
+
+                r = unit_name_mangle_with_suffix(*s, UNIT_NAME_NOGLOB, ".mount", &x);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name: %m");
+                r = strv_consume(&units, x);
+                if (r < 0)
+                        return log_oom();
+        }
+
+        if (units) {
+                res = strv_join(units, " ");
+                if (!res)
+                        return log_oom();
+                fprintf(f, "Before=%1$s\n", res);
+        }
+
+        return 0;
+}
```

Comment:
given that this function is pretty much the same as write_requires_after() now, could you please merge the two and make it take two parameters: one for the fstab field name, and one for the format string to use for formatting the unit file setting? That should make it very easy to support systemd.x-requires=, systemd.x-after=,systemd.x-before= all with the same function.